### PR TITLE
add Time example

### PR DESCRIPTION
**Pull request by maciejbocianski**
Move examples:
https://os.mbed.com/teams/mbed_example/code/time_HelloWorld/
to this repo

Update example code to latest API's

Docs update PR: https://github.com/ARMmbed/mbed-os-5-docs/pull/1248

### CODE REVIEW

File: APIs_Platform/time_HelloWorld/README.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Time example
+
+You can use the time interface to access the Real Time Clock (RTC). The time is set as an offset measured in seconds from the time epoch, which is library specific. The accepted time epoch is the Unix Epoch. An online converter between human readable time and Unix Epoch time is handy, try this one. If the system is not battery powered, then on each reset, the RTC time is reset. Make sure to either provide battery power to keep the time or to reset it each time the device is run.
```
Comment (AnotherButler) [inline APIs_Platform/time_HelloWorld/README.md]: Query: When we say "try this one," what do we mean?

Comment (maciejbocianski) [reply APIs_Platform/time_HelloWorld/README.md]: My bad, I missed links.
Fixed
Also updated description a bit

Comment (maciejbocianski) [reply APIs_Platform/time_HelloWorld/README.md]: @mprse please review the description

File: APIs_Platform/time_HelloWorld/README.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# Time example
+
+The time interface is used to access the Real Time Clock (RTC). The time is set as an offset measured in seconds from the time epoch, which is library specific. In general the accepted time epoch is the [Unix Epoch](https://en.wikipedia.org/wiki/Unix_time). An online converter between human readable time and Unix Epoch time is handy, try [this](https://www.epochconverter.com/) one. If the system is not battery powered then on each power-on (cold reset) the rtc time will be reset. Make sure to either provide battery power to keep the time or to reset it each time the device is run. During software or warm reset (e.g watchdog reset) RTC counts without interrupt.
```
Comment (mprse) [inline APIs_Platform/time_HelloWorld/README.md]: `Make sure to either provide battery power to keep the time or to reset it each time the device is run.` 
I suggest to change:
`Make sure to either provide battery power to keep the time or to set it each time the device is started.`

Comment (maciejbocianski) [reply APIs_Platform/time_HelloWorld/README.md]: updated

**Pull request merged**